packages/nextra-theme-docs/src/components/toc.tsx

```diff
@@ -57,16 +57,29 @@ export function TOC({ toc, filePath }: TOCProps): ReactElement {
   return (
     <div
       className={cn(
-        'nextra-scrollbar _sticky _top-16 _overflow-y-auto _px-4 _pt-6 _text-sm [hyphens:auto]',
-        '_max-h-[calc(100vh-var(--nextra-navbar-height)-env(safe-area-inset-bottom))]'
+        hasHeadings && '_grid _grid-rows-[min-content_1fr_min-content]', // 1fr: toc headings, min-content: title/footer
+        '_sticky _top-[--nextra-navbar-height] _pt-6 _text-sm',
+        '_max-h-[calc(100vh-var(--nextra-navbar-height))]'
       )}
     >
       {hasHeadings && (
         <>
-          <p className="_mb-4 _font-semibold _tracking-tight">
+          <p
+            className={cn(
+              '_mx-4', // use margin instead padding to not have shadow on scrollbar
+              '_font-semibold _tracking-tight',
+              '_pb-2 _shadow-[0_12px_16px_rgb(var(--nextra-bg))] contrast-more:_shadow-none _z-[1]'
```

Comment:
added nice shadow like in footer 
<img width="377" alt="image" src="https://github.com/user-attachments/assets/37b34259-5399-4e96-bf56-58d739c69a1c">